californium-core/src/main/java/org/eclipse/californium/core/CoapServer.java

```diff
@@ -107,7 +107,7 @@
 	private MessageDeliverer deliverer;
 
 	/** The list of endpoints the server connects to the network. */
-	private final List<Endpoint> endpoints;
+	protected List<Endpoint> endpoints;
```

Comment:
Why should this be more exposed than with available methods?

Reply:
Erasing endpoints made by super. To make sure that we use OSCoreEndpoints.

Reply:
```
    public OSCoreServer(NetworkConfig config, int... ports) {
        super(config,ports);
        this.unprotected = new ArrayList<>();
        
        //Erase the endpoints created by super
        this.endpoints = new ArrayList<>();
        this.unprotected = new ArrayList<>();
        
        // create endpoint for each port
        for (int port:ports) {
            addEndpoint(new OSCoreEndpoint(port, config, this.unprotected));
        }
    }
```

do you try to call super with a empty list of ports (just `super(config);`)? 
Then there would be no requirement to remove the useless created endpoints.

If you insist on doing it in your way, can you please ensure, that the already created endpoint are proper released (calling destroy())? (But I would strongly recommend to go for the empty port list).

Reply:
I like your way. I will change it.

Reply:
I miss the remove of this change.

Reply:
Still pending to remove or do you require this change?